### DISCUSSION TITLE
Importing CSV file exported from Firefly - transfers have accounts reversed

### DISCUSSION THREAD

**Discussion by qirtaiba** [firefly-iii/firefly-iii#6418] [Firefly III]
Following on from [this](https://github.com/firefly-iii/firefly-iii/discussions/6416), describing my efforts to migrate Firefly from a Sandstorm instance to a Docker instance, I have given up on trying to transfer the MySQL database, and have gone the CSV import route. However, almost all of the transfers have come up with the accounts reversed.

For example, in my original Firefly installation, here is a transfer transaction:

| Description | Amount | Date | From | To |
| - | - | - | - | - |
| Repayment of mistaken deposit | $436.94 | March 1, 2019 | US Bank | Founder loan | 

Here is how it appears in the CSV export:

`299,602,20190301,"Repayment of mistaken deposit",USD,436.94,,,Transfer,9,"Founder loan",,,,XBT,3,"US Bank",,,,USD,,,,,,,,`

Here is how it imported into the new Firefly:

| Description | Amount | Date | Source account | Destination account |
| - | - | - | - | - |
| Repayment of mistaken deposit | $436.94 | March 1, 2019 | Founder loan | US Bank | 

This also affected all of my foreign currency transactions, with (eg.) a payment of 0.1 XBT (Bitcoins) converted to USD $2000 being represented as 2000 XBT and $0.10.

Fixing all this over 4 years of past transactions is going to be a pain. Is this a bug in the importer that I should report? Or is it a limitation of the export file? Any suggestions on fixing this as a batch?

Thanks.

**Comment by JC5** [comment]
https://github.com/firefly-iii/firefly-iii/issues/6377

**Reply by qirtaiba** [reply]
Awesome, using the development version, the import looks **much** better! Still a few errors and discrepancies  but nothing much.

Thanks!
